apps/vocabulary/main.go

```diff
@@ -0,0 +1,261 @@
+		v := pair.Value
+		if v.Get != nil {
+			printOperation(v.Get, operationId, names)
+		}
+		if v.Post != nil {
+			printOperation(v.Post, operationId, names)
+		}
+		if v.Put != nil {
+			printOperation(v.Put, operationId, names)
+		}
+		if v.Patch != nil {
+			printOperation(v.Patch, operationId, names)
+		}
+		if v.Delete != nil {
+			printOperation(v.Delete, operationId, names)
+		}
+	}
+}
+
+//^^^ Get rid of print post/get/indent
+
+func printOperation(operation *pb.Operation, operationId map[string]int, names map[string]int) {
+	if operation.OperationId != "" {
+		addToMap(operation.OperationId, operationId)
+	}
+	for _, item := range operation.Parameters {
+		switch t := item.Oneof.(type) {
+		case *pb.ParametersItem_Parameter:
+			switch t2 := t.Parameter.Oneof.(type) {
+			case *pb.Parameter_BodyParameter:
+				addToMap(t2.BodyParameter.Name, names)
+			case *pb.Parameter_NonBodyParameter:
+				switch t3 := t2.NonBodyParameter.Oneof.(type) {
+				case *pb.NonBodyParameter_FormDataParameterSubSchema:
+					addToMap(t3.FormDataParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_HeaderParameterSubSchema:
+					addToMap(t3.HeaderParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_PathParameterSubSchema:
+					addToMap(t3.PathParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_QueryParameterSubSchema:
+					addToMap(t3.QueryParameterSubSchema.Name, names)
+				}
+			}
+		}
+	}
+}
+
+func printSchema(schema *pb.Schema, properties map[string]int) {
+	if schema.Properties != nil {
+		for _, pair := range schema.Properties.AdditionalProperties {
+			addToMap(pair.Name, properties)
+		}
+	}
+}
+
+func main() {
+	flag.Parse()
+	args := flag.Args()
+
+	document, err := readDocumentFromFileWithName(args[0])
+
+	if err != nil {
+		log.Printf("Error reading %s. This sample expects OpenAPI v2.", args[0])
+		os.Exit(-1)
+	}
+
+	var schemas map[string]int
+	schemas = make(map[string]int)
+
+	var operationId map[string]int
+	operationId = make(map[string]int)
+
+	var names map[string]int
+	names = make(map[string]int)
+
+	var properties map[string]int
+	properties = make(map[string]int)
+
+	printDocument(document, schemas, operationId, names, properties)
+
+	vocab := &Vocabulary{}
+	for k, v := range schemas {
+		temp := &WordCount{
+			Word:  k,
+			Count: int32(v),
+		}
+		vocab.Schemas = append(vocab.Schemas, temp)
+	}
+
+	for k, v := range operationId {
+		temp := &WordCount{
+			Word:  k,
+			Count: int32(v),
+		}
+		vocab.Operations = append(vocab.Operations, temp)
+	}
+
+	for k, v := range names {
+		temp := &WordCount{
+			Word:  k,
+			Count: int32(v),
+		}
+		vocab.Paramaters = append(vocab.Paramaters, temp)
+	}
+
+	for k, v := range properties {
+		temp := &WordCount{
+			Word:  k,
+			Count: int32(v),
+		}
+		vocab.Properties = append(vocab.Properties, temp)
+	}
+
+	bytes, err := proto.Marshal(vocab)
+	if err != nil {
+		panic(err)
+	}
+	err = ioutil.WriteFile("vocabulary_results.pb", bytes, 0644)
```

Comment:
Suggestion: prefix the generated file with the spec file's name. Makes it easier to use this tool in scripts, because the user wont have to `mv` the generated results file to protect from being overwritten.